design/application-abi.md

```diff
@@ -15,26 +15,42 @@ Current Unstable ABI
 There are two kinds of modules:
 
  - A *command* exports a function named `_start`, with no arguments and no return
-   values. Environments shall call this function once, after instantiating the
-   module and all of its dependencies. After this function exits, the instance
-   is considered terminated and no further use of any of its exports should be
-   permitted.
+   values.
 
- - A *reactor* exports a function named `_initialize`, with no arguments and no
-   return values. Environments shall call this function once, after instantiating
-   the module and all of its dependencies. After this function exits, the instance
-   remains live, and its exports may be accessed.
+   Command instances may assume that they will be called from the environment
+   at most once. Command instances may assume that none of their exports are
+   accessed outside the duraction of that call.
```

Comment:
Is there any case there certain embedders might want to call certain optional auxiliary exports while `_start` is running?  i.e. during a wasi syscall?      I can't think of such a use right now... so maybe this is not a thing?

Reply:
No, and it wouldn't be polyfillable even if it could, because of cyclic dependencies.

Reply:
Sounds reasonable.